Source/Core/Core/PowerPC/GDBThread.cpp

```diff
@@ -0,0 +1,1322 @@
+  DEBUG_LOG(GDB_THREAD, "gdb: write memory: %08x bytes to %08x\n", len, addr);
+
+  hex2mem(addr, m_cmd_bfr + i, len);
+  gdb_reply("OK");
+}
+
+void GDBThread::gdb_read_register(void)
+{
+  if (!m_connected)
+    return;
+
+  u8 reply[64] = { 0 };
+  u32 id = 0;
+
+  gdb_ack();
+
+  u32 i = 1;
+  while (i < m_cmd_len)
+    id = (id << 4) | hex2char(m_cmd_bfr[i++]);
+
+  if (0 <= id && id <= 31)
+  {
+    wbe32hex(reply, GPR(id));
+  }
+  else if (32 <= id && id <= 63)
+  {
+    wbe64hex(reply, riPS1(id - 32));
+    wbe64hex(reply + 16, riPS0(id - 32));
+  }
+  else
+  {
+    switch (id)
+    {
+    case 64:
+      wbe32hex(reply, PC);
+      break;
+    case 65:
+      wbe32hex(reply, MSR);
+      break;
+    case 66:
+      wbe32hex(reply, GetCR());
+      break;
+    case 67:
+      wbe32hex(reply, LR);
+      break;
+    case 68:
+      wbe32hex(reply, CTR);
+      break;
+    case 69:
+      wbe32hex(reply, XER);
+      break;
+    case 70:
+      wbe32hex(reply, 0x0BADC0DE);
+      break;
+    case 71:
+      wbe32hex(reply, FPSCR.Hex);
+      break;
+    default:
+      return gdb_reply("E01");
+      break;
+    }
+  }
+
+  gdb_reply(reinterpret_cast<char *>(reply));
+}
+
+void GDBThread::gdb_write_register(void)
+{
+  if (!m_connected)
+    return;
+
+  u32 id = 0;
+
+  gdb_ack();
+
+  u32 i = 1;
+  while (m_cmd_bfr[i] != '=')
+    id = (id << 4) | hex2char(m_cmd_bfr[i++]);
+  ++i;
+
+  u8 * bufptr = m_cmd_bfr + i;
+
+  if (0 <= id && id <= 31)
+  {
+    GPR(id) = rbe32hex(bufptr);
+  }
+  else if (32 <= id && id <= 63)
+  {
+    riPS1(id - 32) = rbe64hex(bufptr);
+    riPS0(id - 32) = rbe64hex(bufptr + 16);
+  }
+  else
+  {
+    switch (id)
+    {
+    case 64:
+      PC = rbe32hex(bufptr);
+      break;
+    case 65:
+      MSR = rbe32hex(bufptr);
+      break;
+    case 66:
+      SetCR(rbe32hex(bufptr));
+      break;
+    case 67:
+      LR = rbe32hex(bufptr);
+      break;
+    case 68:
+      CTR = rbe32hex(bufptr);
+      break;
+    case 69:
+      XER = rbe32hex(bufptr);
+      break;
+    case 70:
+      // do nothing, we don't have MQ
+      break;
+    case 71:
+      FPSCR.Hex = rbe32hex(bufptr);
+      break;
+    default:
+      return gdb_reply("E01");
+      break;
+    }
+  }
+
+  gdb_reply("OK");
+}
+
+void GDBThread::gdb_handle_query(void)
+{
+  if (!m_connected)
+    return;
+
+  INFO_LOG(GDB_THREAD, "gdb: query '%s'\n", m_cmd_bfr + 1);
+  gdb_ack();
+  gdb_reply("");
+}
+
+void GDBThread::gdb_step(void)
+{
+  if (!m_connected)
+    return;
+
+  gdb_ack();
+
+  if (CPU::State::Stepping != CPU::GetState())
+    return;
+
+  PowerPC::breakpoints.ClearAllTemporary();
+  JitInterface::InvalidateICache(PC, 4, true);
+
+  Common::Event sync_event;
+
+  sync_event.Reset();
+  CPU::StepOpcode(&sync_event);
+
+  sync_event.Wait();
+
+  gdb_signal((u32)SignalTypes::Trap);
+}
+
+void GDBThread::gdb_add_bp(void)
+{
+  if (!m_connected)
+    return;
+
+  u32 type = 0, addr = 0, len = 0, i = 1;
```

Comment:
These should be declared on their own lines. Some of these can also be joined with their first assignment.